### PR TITLE
Implement network management modules

### PR DESCRIPTION
**Pull request by lemoony**
Partly addresses #102 . 

This PR basically allows to create a network and attaching apps to it.

Added two new library files: 
- library/dokku_network.py 
- library/dokku_network_property.py 

Updated README.md

### CODE REVIEW

File: library/dokku_network_property.py
Comment (ltalirz) [inline library/dokku_network_property.py:3]: remove?

File: README.md
Comment (ltalirz) [inline README.md:558]: This seems a bit hackish, maybe have a look at `dokku_domains` and how this switch is handled there?

File: library/dokku_network_property.py
Comment (ltalirz) [inline library/dokku_network_property.py:10]: Could you include an example of clearing a network property?

File: library/dokku_network_property.py
Comment (ltalirz) [inline library/dokku_network_property.py:20]:
```suggestion
      - The value of the network property (leave empty to unset)
```

File: library/dokku_network_property.py
Comment (ltalirz) [inline library/dokku_network_property.py:68]: Is this the right logic?

I'm reading [here](https://dokku.com/docs/networking/network/#attaching-an-app-to-a-network) that one should leave the value empty in order to de-associate the container from the network rather than leave the name of the property empty.

File: library/dokku_network_property.py
Comment (ltalirz) [inline library/dokku_network_property.py:26]: This is just a suggestion, perhaps not the best
```suggestion
      - The network property of the app to be modified
```

File: library/dokku_network_property.py
Comment (ltalirz) [inline library/dokku_network_property.py:18]: Looking at [how `network:set` works](https://dokku.com/docs/networking/network/#attaching-an-app-to-a-network), I think it would be cleaner to call this variable `value`.

The network name just so happens to be the value for the three properties you have chosen to support here

File: library/dokku_network_property.py
Comment (ltalirz) [inline library/dokku_network_property.py:89]: Is there a reason to limit the set of properties to these three?

How about also supporting `static-web-listener` (not sure whether there are more)?properties

File: library/dokku_network_property.py
Comment (ltalirz) [inline library/dokku_network_property.py:40]: let's move the `property` above the value since, conceptually, you first need to know which property you are setting

**Review by ltalirz — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution @lemoony and sorry for the slight delay.

A couple of minor comments before we can get this merged.

**Comment by ltalirz** [comment]
P.S. Would be great to add a short test to the [verify playbook](https://github.com/dokku/ansible-dokku/blob/master/molecule/default/verify.yml).

File: library/dokku_network.py
Comment (lemoony) [reply library/dokku_network.py:3]: Can I ask why? All other modules also have the exact same lines. Also, isn't 

> #!/usr/bin/python

required for Ansible modules?

File: library/dokku_network_property.py
Comment (lemoony) [reply library/dokku_network_property.py:10]: There actually is already an example - please see _"De-associate the container with the network."_

However, I understand that switching between the expressions _"assoication a network"_ and _"setting a network proprety"_ is a bit confusing. However, I tried to stick to the official dokku documentation when it comes to that.

What do you think about changing **"De-associate the container with the network."** to _"Clearing a network property (de-associate the container with the network)."_?

File: library/dokku_network_property.py
Comment (lemoony) [reply library/dokku_network_property.py:68]: Yes, you're completely right. I mixed that up

File: library/dokku_network_property.py
Comment (lemoony) [reply library/dokku_network_property.py:89]: After skimming through code & documentation, i think these properties are also supported:

- bind-all-interfaces
- static-web-listener
- tld

I'm gonna add them.

File: library/dokku_network_property.py
Comment (lemoony) [reply library/dokku_network_property.py:10]: There is  already an example (see _De-associate the container with the network._) but I will rename it to make it more clear

**Comment by lemoony** [comment]
Hi @ltalirz , thanks for the feedback! I already applied most of your suggestions. I will also try to add a test the next couple of days an then request a new review :)

File: library/dokku_network.py
Comment (ltalirz) [reply library/dokku_network.py:3]: Sure, I was just referring to the last line which is a commented import that is likely a leftover from previous edits.
My last PR did clean up the codebase a bit but if you encounter those, please feel free to delete them

File: library/dokku_network_property.py
Comment (ltalirz) [reply library/dokku_network_property.py]: Thanks!

I leave it up to you whether you want to keep the list of properties here explicitly (which means we need to maintain the list in case dokku changes) or whether we simply allow any value and bubble up the error message to the user.

Whatever you decide, it may make sense to mention the list you now collected somewhere in the documentation of the module

**Comment by ltalirz** [comment]
> Hi @ltalirz , thanks for the feedback! I already applied most of your suggestions. I will also try to add a test the next couple of days an then request a new review :)

Thanks a lot, much appreciated!

File: library/dokku_network.py
Comment (lemoony) [reply library/dokku_network.py]: Ah okay, I see. Makes sense.

File: library/dokku_network_property.py
Comment (lemoony) [reply library/dokku_network_property.py]: Yes, I get your point. I think it makes to accept any string and let Dokku do the verification.

File: molecule/default/verify.yml
Comment (ltalirz) [inline molecule/default/verify.yml:110]: I think you want to go with `example-app` here, which is created in the lines above

Comment (lemoony) [reply molecule/default/verify.yml]: Yeah, wasn't the only problem... I had a hard time getting molecule test getting to run on my local system properly. Neverless, tests are working now.

**Comment by lemoony** [comment]
Hi @ltalirz, from my point of view i've addressed every issue. I've added some basic tests for network management. If we want to have more thorough tests, maybe it worth the effort to split `verify.yml` up into multiple files (otherwise it gets to confusing i guess).

If you find the time, would be great to have another review on it. Thanks!

File: molecule/default/verify.yml
Comment (ltalirz) [reply molecule/default/verify.yml]: > I had a hard time getting molecule test getting to run on my local system properly. Neverless, tests are working now.

Ah - I mostly just rely on the CI test setup myself.
What are the issues you ran into? Would be happy to add them to the [contributing docs](https://github.com/dokku/ansible-dokku/blob/master/CONTRIBUTING.md) in a separate PR.

File: library/dokku_network_property.py
Comment (ltalirz) [inline library/dokku_network_property.py:72]: I suggest you check whether both `app` and `global` attributes are set and return an error if they are.

My feeling is that constructing these command strings with if blocks step by step is easier to understand and maintain, but feel free to keep this more compact notation.

File: library/dokku_network_property.py
Comment (ltalirz) [inline library/dokku_network_property.py:93]: Property is required, isn't it?

```suggestion
        "property": {"required": True, "type": "str"},
```

File: library/dokku_network_property.py
Comment (ltalirz) [inline library/dokku_network_property.py:94]: Value isn't required

```suggestion
        "value": {"required": False, "type": "str"},
```

Please just add one more task to the verify playbook that unsets a property (which would have caught this)

**Review by ltalirz — CHANGES_REQUESTED** [review]
thanks a lot @lemoony 

almost ready to merge

File: library/dokku_network_property.py
Comment (lemoony) [reply library/dokku_network_property.py:75]: `data["value"]` will only be executed by the engine if the condition `"value" in data` evalutes to true, otherwise the empty string value will be taken. I just tested this with a small python script and it should actually be fine.

Comment (lemoony) [reply library/dokku_network_property.py:75]: Ahh sorry, I just realized what you may have pointed at: The above solution works if `value` is not present at all. However, if `value`is set to `None` the expression actually evaluates to true. 

I have to fix that. Thanks for spotting that!

File: molecule/default/verify.yml
Comment (lemoony) [reply molecule/default/verify.yml]: The thing is that molecule does not tell my on my local setup what's actually wrong:

I```
NFO     Running default > verify
INFO     Running Ansible Verifier

PLAY [Verify] ******************************************************************

TASK [Gathering Facts] *********************************************************
ok: [instance]

TASK [Run role without running any tasks] **************************************
ERROR! Could not find specified file in role: tasks/init.yml

PLAY RECAP *********************************************************************
instance                   : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

CRITICAL Ansible return code was 2, command was: ansible-playbook --inventory /Users/lemoony/.cache/molecule/ansible-dokku/default/inventory --skip-tags molecule-notest,notest /Users/lemoony/projects/opensource/ansible-dokku/molecule/default/verify.yml
WARNING  An error occurred during the test sequence action: 'verify'. Cleaning up.
INFO     Running default > cleanup
WARNING  Skipping, cleanup playbook not configured.
INFO     Running default > destroy
```

So I have to push the code each time to see what's wrong...

**Review by ltalirz — APPROVED** [review]
thanks, all looking good now.

I see the tests pass on your fork - will merge

**Comment by ltalirz** [comment]
And dropped new release https://github.com/dokku/ansible-dokku/releases/tag/v2021.9.3

**Pull request merged**
